### PR TITLE
fix: disable adding execution tags when lambda is not traced

### PR DESCRIPTION
**Pull request by vladlLumigo**

### CODE REVIEW

File: src/globals.ts [hunk 1]
```diff
@@ -95,10 +101,15 @@ export const ExecutionTags = (() => {
 
   const addTag = (key, value, shouldLogErrors = true) => {
     try {
-      logger.debug(`Adding tag: ${key} - ${value}`);
-      if (!validateTag(key, value, shouldLogErrors)) return false;
-      // @ts-ignore
-      global.tags.push({ key: normalizeTag(key), value: normalizeTag(value) });
+      if (isLambdaTraced()) {
+        logger.debug(`Adding tag: ${key} - ${value}`);
+        if (!validateTag(key, value, shouldLogErrors)) return false;
```
Comment (therightstuff) [inline src/globals.ts:106]: please use braces (`{`, `}`) even for single lines

File: src/globals.ts [hunk 2]
```diff
@@ -95,10 +101,15 @@ export const ExecutionTags = (() => {
...
+        // @ts-ignore
```
Comment (therightstuff) [inline src/globals.ts:107]: why do we need this `@ts-ignore`? can we put the reason inline?

File: src/globals.ts [hunk 3]
```diff
@@ -95,10 +101,15 @@ export const ExecutionTags = (() => {
...
+        global.tags.push({ key: normalizeTag(key), value: normalizeTag(value) });
+      } else {
+        shouldLogErrors && logger.warnClient(`${ADD_TAG_ERROR_MSG_PREFIX}: lambda is not traced`);
+        return false;
+      }
     } catch (err) {
       shouldLogErrors && logger.warnClient(ADD_TAG_ERROR_MSG_PREFIX);
```
Comment (therightstuff) [inline src/globals.ts:114]: do you know why the error isn't included in the client warning?

File: src/utils.ts [hunk 4]
```diff
@@ -586,3 +586,7 @@ export const filterObjectKeys = (
     .reduce((cur, key) => {
       return Object.assign(cur, { [key]: obj[key] });
     }, {});
+
+export const isKillSwitchOn = () => !!process.env['LUMIGO_SWITCH_OFF'];
```
Comment (therightstuff) [inline src/utils.ts:590]: please verify that this is correct, if i recall correctly this should be a string comparison with very specific cases where it's "true"

File: src/utils.ts [hunk 5]
```diff
@@ -586,3 +586,7 @@ export const filterObjectKeys = (
...
+
+export const isLambdaTraced = () => !!(!isKillSwitchOn() && isAwsEnvironment());
```
Comment (therightstuff) [inline src/utils.ts:592]: both of these methods should return a boolean value, so we should not need `!!`. for increased readability, please put the positive check first (ie. `=> (isAwsEnvironment() && !isKillSwitchOn())`)

**Review by therightstuff — CHANGES_REQUESTED** [review]
i do have a few concerns i've highlighted, but it's mostly great :)

File: src/utils.ts (hunk 4 above)
Comment (doriaviram) [inline src/utils.ts:590]: There is a method for that `isSwitchedOff`

File: src/globals.ts (hunk 2 above)
Comment (vladlLumigo) [reply src/globals.ts:107]: Good question, it was in the original code, I don't know what was the reason

File: src/globals.test.js [hunk 6]
```diff
@@ -445,4 +453,26 @@ describe('globals', () => {
...
+  test.each`
+    killSwitchValue | isAwsEnvironment | expectedRetValue | expectedTags
+    ${false}        | ${true}          | ${true}          | ${[{ key: 'k0', value: 'v0' }]}
+    ${true}         | ${true}          | ${false}         | ${[]}
+    ${true}         | ${false}         | ${false}         | ${[]}
+    ${false}        | ${false}         | ${false}         | ${[]}
+  `(
+    'killSwitch=$killSwitchValue, isAwsEnv=$isAwsEnvironment, expectedRetValue=$expectedRetValue$',
+    ({ killSwitchValue, isAwsEnvironment, expectedRetValue, expectedTags }) => {
+      const oldEnv = Object.assign({}, process.env);
```
Comment (doriaviram) [inline src/globals.test.js:466]: You don't need to save the envs. the infra do it for you.
Look at steup.js

File: src/globals.test.js [hunk 7]
```diff
@@ -445,4 +453,26 @@ describe('globals', () => {
 
     process.env = { ...oldEnv };
   });
+
+  test.each`
```
Comment (doriaviram) [inline src/globals.test.js:457]: It looks like its hard to maintain these spaces, maybe array its better

**Review by doriaviram — CHANGES_REQUESTED** [review]

File: src/globals.ts (hunk 1 above)
Comment (vladlLumigo) [reply src/globals.ts:106]: done

File: src/utils.ts (hunk 4 above)
Comment (vladlLumigo) [reply src/utils.ts:590]: used already existing function isSwitchedOff() instead

File: src/utils.ts (hunk 5 above)
Comment (vladlLumigo) [reply src/utils.ts:592]: done

File: src/utils.ts (hunk 4 above)
Comment (vladlLumigo) [reply src/utils.ts:590]: done

File: src/globals.test.js (hunk 6 above)
Comment (vladlLumigo) [reply src/globals.test.js:466]: done

File: src/globals.test.js (hunk 7 above)
Comment (vladlLumigo) [reply src/globals.test.js:457]: tried both versions,  decided to stay with the table version

**Review by therightstuff — APPROVED** [review]

File: src/globals.ts (hunk 3 above)
Comment (vladlLumigo) [reply src/globals.ts:114]: we show the client only specific errors, a general catch , will contain information that the client don't really need to see

**Review by doriaviram — APPROVED** [review]

**Review by doriaviram — APPROVED** [review]

**Pull request merged**